src/Transport/RestTransport.php

```diff
@@ -109,24 +111,34 @@ public static function build(string $apiEndpoint, string $restConfigPath, array
      */
     public function startUnaryCall(Call $call, array $options)
     {
+        $requestLog = null;
         $headers = self::buildCommonHeaders($options);
+        $request =  $this->requestBuilder->build(
+            $call->getMethod(),
+            $call->getMessage(),
+            $headers
+        );
+
+        if ($this->logger) {
+            $requestLog = $this->logRequest($request);
+        }
 
         // call the HTTP handler
         $httpHandler = $this->httpHandler;
         return $httpHandler(
-            $this->requestBuilder->build(
-                $call->getMethod(),
-                $call->getMessage(),
-                $headers
-            ),
+            $request,
             $this->getCallOptions($options)
         )->then(
-            function (ResponseInterface $response) use ($call, $options) {
+            function (ResponseInterface $response) use ($call, $options, $requestLog) {
                 $decodeType = $call->getDecodeType();
                 /** @var Message $return */
                 $return = new $decodeType;
                 $body = (string) $response->getBody();
 
+                if ($this->logger) {
+                    $this->logResponse($response, $requestLog['timestamp']);
```

Comment:
Why not just pass the timestamp in, instead of passing in the whole `$requestLog`? Likewise, why not just have `requestLog` return the timestamp instead of the whole log? Or even better, get the timestamp in `startUnaryCall`, and pass the timestamp into `requestLog` instead of getting it back from that function.

Reply:
Comment for future reference:
We decided to change the logic for the latency to be calculated on the `loggerMiddleware`